source/opt/instruction.h

```diff
@@ -550,10 +552,28 @@ class Instruction : public utils::IntrusiveNodeBase<Instruction> {
   // OpenCLDebugInfo100InstructionsMax.
   OpenCLDebugInfo100Instructions GetOpenCL100DebugOpcode() const;
 
+  // Returns debug opcode of an NonSemantic.Vulkan.DebugInfo.100 instruction. If
+  // it is not an NonSemantic.Vulkan.DebugInfo.100 instruction, just returns
+  // NonSemanticVulkanDebugInfo100InstructionsMax.
+  NonSemanticVulkanDebugInfo100Instructions GetVulkan100DebugOpcode() const;
+
+  // Returns debug opcode of an OpenCL.100.DebugInfo or
+  // NonSemantic.Vulkan.DebugInfo.100 instruction. Since these overlap, we
+  // return the OpenCLDebugInfo code
+  CommonDebugInfoInstructions GetCommonDebugOpcode() const;
+
   // Returns true if it is an OpenCL.DebugInfo.100 instruction.
   bool IsOpenCL100DebugInstr() const {
     return GetOpenCL100DebugOpcode() != OpenCLDebugInfo100InstructionsMax;
   }
+  // Returns true if it is an NonSemantic.Vulkan.DebugInfo.100 instruction.
```

Comment:
You have done this in a number of places.  Is "an NonSemantic instruction" correct?  I thought it should be "a NonSemantic instruction."

Reply:
Fixed